online-shop/src/sagas/index.tsx

```diff
@@ -1,29 +1,56 @@
 import { all, takeLatest, put, call } from "redux-saga/effects";
-import { FetchSelectedProduct } from "../actions/productActions";
+import {
+  FetchSelectedProduct,
+  GET_SELECTED_PRODUCT,
+  DELETE_CURRENT_PRODUCT,
+  changeLoadingIndicator,
+  loadProduct
+} from "../actions/productActions";
 import { ICartProduct } from "../model/Interfaces";
-import { FetchOrder } from "../actions/shoppingActions";
-import { FetchUpdateProduct, FetchAddNewProduct } from "../actions/editActions";
+import {
+  FetchOrder,
+  SEND_ORDER,
+  switchLoading,
+  checkOut
+} from "../actions/shoppingActions";
+import {
+  FetchUpdateProduct,
+  FetchAddNewProduct,
+  UPDATE_CURRENT_PRODUCT,
+  CREATE_NEW_PRODUCT
+} from "../actions/editActions";
+import {
+  GET_PRODUCT_LIST,
+  changeLoadingStatus,
+  deleteProductFromUI,
+  loadListIncognito,
+  loadProducts,
+  fetchTheList
+} from "../actions/productListActions";
+import {
+  GET_SALES,
+  swithLoadingStatus,
+  loadCurrentSales
+} from "../actions/highChartActions";
 
 function* fetchProductList() {
   const data = yield fetch("http://localhost:4000/products", { method: "GET" })
     .then(response => response.json())
     .then(result => {
       return result;
     });
-  yield put({ type: "LOAD-PRODUCTS", data });
+  yield put(loadProducts(data, true));
+  yield put(changeLoadingStatus());
 }
 
 function* fetchDeleteDetailsProduct(action: FetchSelectedProduct) {
   yield fetch("http://localhost:4000/products/" + action.productID, {
```

Comment:
I would recommend extracting the fetch calls (+ corresponding then / catch) into service / API objects in a separate folder and just call these service methods here in the sagas. That way the saga looks cleaner.